src/engine.rs

```diff
@@ -503,7 +522,8 @@ impl evm::backend::Backend for Engine {
 
     /// Get storage value of address at index.
     fn storage(&self, address: Address, index: H256) -> H256 {
```

Comment:
Not from this PR, but still

It is weird to use `index` for this variable name here and `key` elsewhere (see `get_storage`/`set_storage`), when they are referring to the same thing (have same semantics). I'd keep `key` which I believe is what ethereum is calling this value.

Reply:
Yes, `SSTORE` and `SLOAD` [use](https://ethervm.io) the nomenclature `key`.